modules/monitoring-configuring-remote-write.adoc

```diff
@@ -0,0 +1,176 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+[id="configuring_remote_write_storage_{context}"]
+= Configuring remote write storage
+
+You can configure remote write to enable Prometheus to send ingested metrics to remote systems for storage. 
+Doing so has no impact on how or for how long Prometheus stores metrics using local storage.
+
+.Prerequisites
+
+* *If you are configuring core OKD monitoring components:*
+** You have access to the cluster as a user with the `cluster-admin` role.
+** You have created the `cluster-monitoring-config` `ConfigMap` object.
+* *If you are configuring components that monitor user-defined projects:*
+** You have access to the cluster as a user with the `cluster-admin` role or as a user with the `user-workload-monitoring-config-edit` role in the `openshift-user-workload-monitoring` project.
+** You have created the `user-workload-monitoring-config` `ConfigMap` object.
+* You have installed the OpenShift CLI (`oc`).
+* You have set up a remote write compatible endpoint (such as Thanos) and know the endpoint URL.
+* You have set up authentication credentials for the remote write endpoint. 
++
+[CAUTION]
+====
+To reduce security risks, avoid sending metrics to an endpoint via unencrypted HTTP or without using authentication.
+====
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add a `remoteWrite:` section under `data/config.yaml/prometheusK8s`.
+
+. In this section, add an endpoint URL, authetication credentials for the endpoint, and write label configurations.
+
+.. Add an endpoint URL and authentication credentials:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote.write.endpoint"
+        <endpoint_authentication_credentials>
+----
++
+For `endpoint_authentication_credentials` substitute your credentials.
```

Comment:
Should we enumerate these better to list the types of authentication mechanisms we support? We list Basic Authentication and TLS authentication in examples but are these the only ones?

Reply:
Should Authentication be its own 'section'? cc: @rolfedh

Reply:
Let's see how the proposed updates to the structure work. If they don't disrupt the flow too much, it might be better for users to deliver just-in-time information this way.

Reply:
@sferich888 - according to @jan--f, TLS Authentication and Basic Authentication are the only 2 protocols we are supporting. We should add a sample with values for tlsConfig. Jan, can you please provide a sample?

Reply:
Added samples for both auth methods.